modules/tinymce/src/themes/silver/main/ts/modes/Inline.ts

```diff
@@ -108,19 +110,34 @@ const render = (editor: Editor, uiComponents: RenderUiComponents, rawUiConfig: R
     editor.nodeChanged();
   };
 
-  editor.on('focus show', render);
-  editor.on('blur hide', ui.hide);
+  editor.on('show', render);
+  editor.on('hide', ui.hide);
+
+  if (!isToolbarPersist(editor)) {
+    editor.on('focus', render);
+    editor.on('blur', ui.hide);
+  }
 
   editor.on('init', () => {
-    if (editor.hasFocus()) {
+    if (editor.hasFocus() || isToolbarPersist(editor)) {
```

Comment:
A thought: if this setting is only to be used on init, and people should use API methods to programmatically change the state, couldn't we cache the result of `isToolbarPersist(editor)`? Tiny optimisation, but it'd be a few less calls and it would mean people couldn't do weird things like change the value of the setting at runtime.